### DISCUSSION TITLE
How can I use router in store

### DISCUSSION THREAD

**Discussion by knut9900848** [quasarframework/quasar#11803] [CLI - PWA mode] [answered]
I'm trying to redirect to dashboard after login. 
I found this in app.js file that

` 
// make router instance available in store
store.$router = router
`

So I codede that in my login method of auth.js file like below.

`store.$router.push({ name: 'dashboard' })`

But there is nothing to happen.

How can I use router in vuex store file?

**Comment by metalsadman** [comment, marked as answer]
use `function` on your store actions, then you can access it via `this`. 
ie. 
```js
// some actions.js
export function login(...) {
  ...
  this.$store.push(...)
  ...
}
```
see other methods here https://forum.quasar-framework.org/topic/3960/access-router-outside-vue/9.
